.github/workflows/coverage-report.yml

```
@@ -0,0 +1,95 @@
+      matrix:
+        package: ${{ fromJson(needs.report.outputs.matrix) }}
+
+    steps:
+      # We need to re-download the artifact in each matrix job
+      - name: Download Individual Coverage Reports
+        uses: actions/download-artifact@v4
+        with:
+          name: individual-coverage-data # Match upload name
+          path: coverage-artifacts
+          github-token: ${{ secrets.GITHUB_TOKEN }}
+          run-id: ${{ needs.report.outputs.run_id || github.event.workflow_run.id }} # Use run_id from needs if possible
```

actionlint (1.7.4)

property "run_id" is not defined in object type {matrix: string}

(expression)